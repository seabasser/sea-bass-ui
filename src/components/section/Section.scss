@import '../../utils/_partials.scss';

.section {
  @include dark-theme;
  @include light-theme;
}