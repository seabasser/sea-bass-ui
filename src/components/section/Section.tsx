import React, { ReactNode } from 'react';
import classNames from 'classnames';

import './Section.scss';

interface MainProps {
  className?: string,
  theme?: 'dark' | 'light',
  children: ReactNode,
}

const Section: React.FC<MainProps> = ({ className, theme, children, ...props }) => {

  const classes = classNames(
    'section',
    `section__theme-${theme}`,
    className
  );

  return (
    <section className={classes} {...props}>{children}</section>
  );
};

Section.defaultProps = {
  theme: 'light'
};

export default Section;

