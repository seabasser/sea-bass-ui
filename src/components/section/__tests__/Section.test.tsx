import React from 'react';
import { render } from '@testing-library/react';
import Section from '../Section';

describe('<Section /> spec', () => {
  it('Renders normally', () => {
    const { asFragment } = render(<Section>test</Section>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders with dark theme', () => {
    const { asFragment, getByText } = render(<Section theme='dark'>test</Section>);
    expect(getByText('test')).toHaveClass('section__theme-dark');
    expect(asFragment()).toMatchSnapshot();
  });

  it('Renders with className', () => {
    const { asFragment, getByText } = render(<Section className='testClass'>test</Section>);
    expect(getByText('test')).toHaveClass('testClass');
    expect(asFragment()).toMatchSnapshot();
  });
});