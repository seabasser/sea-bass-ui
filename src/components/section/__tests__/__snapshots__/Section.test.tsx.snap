// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> spec Renders normally 1`] = `
<DocumentFragment>
  <section
    class="section section__theme-light"
  >
    test
  </section>
</DocumentFragment>
`;

exports[`<Section /> spec Renders with className 1`] = `
<DocumentFragment>
  <section
    class="section section__theme-light testClass"
  >
    test
  </section>
</DocumentFragment>
`;

exports[`<Section /> spec Renders with dark theme 1`] = `
<DocumentFragment>
  <section
    class="section section__theme-dark"
  >
    test
  </section>
</DocumentFragment>
`;
